### ISSUE TITLE
Booking

### ISSUE THREAD

**Issue by polash62** [Travelport-Ukraine/uapi-json#450]
Every booking request I can see following error in response.
`{
"": "Some of the requested AirPricingInfos could not be saved for the requested provider.",
"Code": "0",
"Type": "Warning",
"ProviderCode": "1G"
},
{
"": "The following attribute is missing in the AirCreateReservationReq: AirPricingSolution/HostToken.",
"Code": "0",
"Type": "Error",
"ProviderCode": "1G"
}`

I am using trial credential. I have also tried with different segment of data with different source destination. Every time I have got above error for booking response.
For that reason, when I call ticket function it give me response like "Unable to ticket without pricing information." 

But I can see pricing information and plating carrier is attached in booking request.
In AirCreateReservationReq I can see PlatingCarrier is attached in AirPricingInfo.

<air:AirPricingInfo Key="sogRc77Q2BKAYyz+AAAAAA==" TotalPrice="AUD172.60" BasePrice="USD110.00" ApproximateTotalPrice="AUD172.60" ApproximateBasePrice="AUD163.00" EquivalentBasePrice="AUD163.00" ApproximateTaxes="AUD9.60" Taxes="AUD9.60" LatestTicketingTime="2019-12-15T23:59:00.000+11:00" PricingMethod="Guaranteed" Refundable="true" IncludesVAT="false" ETicketability="Yes" PlatingCarrier="HR" ProviderCode="1G">

AirPricingSolution is also attached in AirCreateReservationReq.

<air:AirPricingSolution Key="sogRc77Q2BKAWyz+AAAAAA==" TotalPrice="AUD172.60" BasePrice="USD110.00" ApproximateTotalPrice="AUD172.60" ApproximateBasePrice="AUD163.00" EquivalentBasePrice="AUD163.00" Taxes="AUD9.60" Fees="AUD0.00" ApproximateTaxes="AUD9.60" QuoteDate="2019-12-08" >

Thanks in Advance.

**Comment by Yeab13** [comment]
Hello polash62,

You can fix that issue by changing these lines from **AirParser.js**

`['air:AirSegment', 'air:AirPricingInfo', 'air:FareNote'].forEach((root) => {  `

to 

`['air:AirSegment', 'air:AirPricingInfo', 'air:FareNote', `common_${this.uapi_version}:HostToken`].forEach((root) => {`

**Comment by polash62** [comment]
Hi Yeab13,

 Thank you.  Problem has been solved by modifying following line as you mentioned. 

`  ['air:AirSegment', 'air:AirPricingInfo', 'air:FareNote', `common_${this.uapi_version}:HostToken`].forEach((root) => {
    const builder = new xml2js.Builder({
      headless: true,
      rootName: root,
    });`

Finally, when I tried to call ticket function it give me following message. 
"Host error during ticket issue. INVALID ND LINKAGE".
I know, ticket functionality is not available for trial credentials. Possibly for that reason, I am getting this message.

**Comment by creduo** [comment]
The missing HostToken issue was fixed and merged in PR #422, maybe need to update uapi-json?

**Comment by polash62** [comment]
Hi Creduo,

I have checked it by manually changing AirParser.js (node_modules\uapi-json\src\Services\Air\AirParser.js) file. 
I have run  "npm update uapi-json", but It does not update my node_modules (node_modules\uapi-json\src\Services\Air\AirParser.js) AirParser.js file part, that needs to be updated. 
I have tried with npm cache clean or npm cache verify then run 'npm install' but did not update uapi-json AirParser.js. 

Thanks,
